### PR TITLE
Small fix for BranchPythonOperator Operator

### PR DESCRIPTION
**Pull request by SandroLuck**
Current behavior: When you Branch from A  e.g. the BranchPythonOperator '->' (B or C,), and you make some B '->' C as well. The current behavior is that C will be skipped even though it's a downstream task of B.
To avoid this behavior this fix will check that only tasks which are not in the downstream of the branch are skipped. I think using this behavior the BranchyPythonOperator is more useful and behaves as expected.

Make sure you have checked _all_ steps below.

### JIRA
- [x] My PR addresses the following [Airflow JIRA](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
    - https://issues.apache.org/jira/browse/AIRFLOW-2609
    - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a JIRA issue.

### Description
- [x] Here are some details about my PR, including screenshots of any UI changes:
No UI Changes description as above

### Tests
- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
Same tests should apply

### Commits
- [x] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### Documentation
- [x] In case of new functionality, my PR adds documentation that describes how to use it.
    - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.

### Code Quality
- [x] Passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

**Comment by SandroLuck** [comment]
Let me know if you need something else for this commit. It's a very small change

**Comment by SandroLuck** [comment]
I'm going to say these Travis tests are failing because of issues not related to my code. Can anyone check out why this is happening? It mentions some aws error. AWS is outside of my control

**Comment by SandroLuck** [comment]
Let me please know why this failing and how I could fix it

**Comment by Noremac201** [comment]
Looks like its failing because your code is not passing the `git diff upstream/master -u -- "*.py" | flake8 --diff` requirement.

airflow/operators/python_operator.py:115:65: W291 trailing whitespace
airflow/operators/python_operator.py:114:70: W291 trailing whitespace
airflow/operators/python_operator.py:113:67: W291 trailing whitespace
airflow/operators/python_operator.py:112:62: W291 trailing whitespace
airflow/operators/python_operator.py:111:70: W291 trailing whitespace

File: airflow/operators/python_operator.py [hunk 1]
```diff
@@ -108,10 +108,11 @@ class BranchPythonOperator(PythonOperator, SkipMixin):
...
     the task_id to follow. The task_id returned should point to a task
     directly downstream from {self}. All other "branches" or
-    directly downstream tasks are marked with a state of ``skipped`` so that
-    these paths can't move forward. The ``skipped`` states are propageted
-    downstream to allow for the DAG state to fill up and the DAG run's state
-    to be inferred.
+    directly downstream tasks are marked with a state of ``skipped``, 
+    unless the downstream tasks are also a downstream task of 
+    the task_id to follow, so that these paths can't move forward. 
+    The ``skipped`` states are propageted downstream to allow for the
```
Comment (Noremac201) [inline airflow/operators/python_operator.py]: propagated

Comment (SandroLuck) [reply airflow/operators/python_operator.py]: Thanks for the feedback will fix

**Comment by Fokko** [comment]
Thanks @SandroLuck 

Could you squash your commits into one single commit?

**Comment by SandroLuck** [comment]
New commit made with one single commit instead of the many I made here https://github.com/apache/incubator-airflow/pull/3530

**Comment by ashb** [comment]
Closing in favour of your other PR

**Pull request closed without merging**